### ISSUE TITLE
Modals, popovers, tribute containers are missing in Fullsreen mode

### ISSUE THREAD

**Issue by Antreesy** [nextcloud/spreed#9299]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Toggle fullscreen mode (in chat or a call)
2. As moderator - try to click "Leave call"
3. If breakout rooms configured - try to open any modal inside Breakout rooms tab

### Expected behaviour
![image](https://user-images.githubusercontent.com/93392545/231514532-d53dc2f3-80da-417e-aca4-ff48e5c36001.png)

### Actual behaviour
![image](https://user-images.githubusercontent.com/93392545/231514704-280447c4-1b44-4571-8789-9398d6de2423.png)

### Possible solution
`this.$store.getters.getMainContainerSelector()` always returns `#content-vue`
In some places we are trying to access this DOM Element only before it is mounted on a page.
Outcome: broken modals outside of #content-vue

## Talk app

**Talk app version:** 27 (could be backported for 26 too)

**Comment by danxuliu** [comment]
This issue rings a bell. I have not checked if it is the same, but sounds like https://github.com/nextcloud/spreed/pull/6879 with different components. If you find a way to properly solve it please apply the fix as well to the file picker and the viewer to get rid of the hack :-) Thanks!

**Comment by nickvergessen** [comment]
The thing is we have a container option but seems we provide it too late or fixed or whatever and therefore it's not remapped to the fullscreen item.
I think Maksim found more details already, but I asked him to create the ticket early so we don't forget

**Comment by ShGKme** [comment]
> The thing is we have a container option but seems we provide it too late or fixed or whatever and therefore it's not remapped to the fullscreen item.

The problem is that in many cases we provide it on the creation moment while it is exist only after rendering and mounting. 

---

I'd try to solve the problem in a `@nextcloud/vue` lib:
- `#content-vue` is  `@nextcloud/vue/NcContent` component
- Tribute is used in the `@nextcloud/vue/NcRichContenteditable` component
- Modals are `@nextcloud/vue/NcDialog` components

So all the time it is a problem about using some floating `@nextcloud/vue` component inside another `@nextcloud/vue` component.

Also it is related to a problem with different `box-sizing` in different Nextcloud page parts.

With a dialogs/viewer it could be more tricky.

**Comment by Antreesy** [comment]
Yes, most of the time we try to append a new modal or tribute into <div #content-vue>, but it's unavailable at the moment before mounting, and we end up each time with appending into <body>.
Slightly different problem with Viewer, it's rendered in NcModal, so container could be passed as a prop (but it has to be fixed in the Viewer repository)
